### PR TITLE
Andrea new

### PR DESCRIPTION
**Pull request by andreamola**
There are a bunch of things added here:
1) pi-BEM dependence on deal2toolkit has been removed;
2) the class SingularKernelIntegrals has been added. It provides functions for the computation of hypersingular and strongly singular integrals;
3) the method compute_gradients_hypersingular has been added. It computes the potential gradient making use of the hypersingular BIE
4) to properly write the hypersingular BIE a method for the computation of the free coefficients in such equation has been also added
5) some tests for the SingularKernelIntegrals class functions have been added

To activate the hypersingular gradient computation run one of the classical examples, but make sure to use

bem.compute_gradients_hypersingular(phi, dphi_dn);

instead of 

bem.compute_gradients_hypersingular(phi, dphi_dn);

in the compute_errors method of boundary_conditions.cc (line 352)

Things that might be improved:
- quadrature order in SingularKernelIntegrals is now hard coded.
- accuracy is lower than gradient recovery, but gradient recovery would never work on the wake of an airfoil, that is why I implemented it
- I don't know if it is worth it to add optional parameters in the param file, to decide whether gradients should be computed in one way or another. I guess it should be left to the library users to call one method or the other in their executable file

Finally. I compiled everything on the current milestone, which is deal.ii 9.5.1 .
I would prefer if we use milestones rather than the git version.

### CODE REVIEW

**Comment by nicola-giuliani** [comment]
@luca-heltai can you help us to fix the non passing checks?

**Comment by luca-heltai** [comment]
I don't see any pending request. I think we need to move to github actions.

File: Doxyfile.in [hunk 1]
```diff
@@ -32,7 +32,7 @@ DOXYFILE_ENCODING      = UTF-8
 # title of most generated pages and in a few other places.
 # The default value is: My Project.
 
-PROJECT_NAME           = "pi-BEM: a parallel BEM solver for deal.II"
+PROJECT_NAME           = "Heat Shell: a heat transmission model for shells in deal.II"
```
Comment (luca-heltai) [inline Doxyfile.in:36]: Why?

**Review by luca-heltai — APPROVED** [review]

**Comment by luca-heltai** [comment]
Modulo the name change, I think we can start merging this, and then we move the testing infrastructure on github actions. 
This should:
- test any PR
- build a docker image with the latest version and publish it

**Pull request merged**